runtime/near-evm-runner/src/utils.rs

```diff
@@ -315,3 +735,36 @@ pub fn ecrecover_address(hash: &RawHash, signature: &[u8; 65]) -> Address {
     }
     Address::zero()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_method() {
+        let methods =
+            MethodAndTypes::parse("Blah(Foo q,Bar w)Foo(uint256 x,Bar z)Bar(T ttt)T(string x)")
```

Comment:
Maybe I'm misunderstanding what this is for, but it seems incorrectly named: it's not a method type, it's a struct type, right?

And if it's a struct type, then [EIP-712](https://eips.ethereum.org/EIPS/eip-712) mandates alphabetical ordering for referenced types (`Bar` before `Foo`):

> If the struct type references other struct types (and these in turn reference even more struct types), then the set of referenced struct types is collected, sorted by name and appended to the encoding. An example encoding is `Transaction(Person from,Person to,Asset tx)Asset(address token,uint256 amount)Person(address wallet,string name).`

Reply:
Yes method_name is not very accurate, I just inherit it from Illia's implementation

Reply:
Name it method_signature now

Reply:
Still trying to think of a better name for it. Thought it can be method_signature. But in functional programming world, signature is something like "adopt(uint)" (without arg name), see above comment eip-712 requires adopt come with arg name, therefore method_signature is inappropriate

Reply:
I name it "method_def" now, to explictly differentiate with method_sig

Reply:
@ailisp Please see my other comment just now. EIP-712 is about structure types, the ABI is about function calls. The structure signature includes the structure's property names; the function signature does _not_ include the function's parameter names, only their types.

An ABI function signature `adopt(uint256)` has nothing to do with EIP-712: it describes a function to call, not a structure type. Conversely, EIP-712 has nothing to do with function or method signatures such as `adopt()`. Hence why the naming here is problematic, and renaming it to `method_def` or `method_sig` will not fix that problem.

Reply:
Your comments is mostly correct, but this is not:

> An ABI function signature adopt(uint256) has nothing to do with EIP-712: it describes a function to call, not a structure type. 

It's not because we need to eip-712 adopt(uint256), but because we need to eip-712 this:
```
NearTx(...)Arguments(uint256 petId)
```
And ^ `uint256 petId` is taken from `adopt(uint256)`

Reply:
OK, after your comments I believe I understand what we're doing. Thanks, @ailisp.